Plugins/Notifiers/bark.md

```diff
@@ -0,0 +1,55 @@
+---
+title: bark
+description: 
+published: true
+date: 2022-10-09T01:49:57.874Z
+tags: 
+editor: markdown
+dateCreated: 2022-10-09T01:49:57.874Z
+---
+
+# [Notifiers](/Plugins/Notifiers) > Bark
+> Bark is a part of the [notifier](/Plugins/Notifiers) plugin system.
+{.is-success}
+
+This plugin allows Flexget to send notifications to a [Bark](https://github.com/Finb/bark-server) client via the bark [API_V2](https://github.com/Finb/bark-server/blob/master/docs/API_V2.md).
+
+## Configuration
+
+| Option |Type|  Description | Default |
+| --- | --- | --- | --- |
+| **server** | string | URL of the Bark server. **Required**
+| **device_key**| string | Key of the device. **Required**
+| level | string | `'active'`, `'timeSensitive'`, or `'passive'`. |
+| badge | integer | The number displayed next to App icon. ([Apple Developer](https://developer.apple.com/documentation/usernotifications/unnotificationcontent/1649864-badge)) |
+| automaticallyCopy | string | Must be `1`. |
+| copy | string |  The value to be copied. |
+| sound | string | Value from [here](https://github.com/Finb/Bark/tree/master/Sounds). |
+| icon | string | An url to the icon, available only on iOS 15 or later. |
+| group | string | The group of the notification. |
+| isArchive | string | Value must be `1`. Whether or not should be archived by the app. |
```

Comment:
We don't use camelCase for options

Reply:
> We don't use camelCase for options

Is this schema acceptable?
```
schema = {
        'type': 'object',
        'properties': {
            'server': {'type': 'string'},
            'device_key': {'type': 'string'},
            'options': {
                'type': 'object',
                'additionalProperties': {
                    'oneOf': [{'type': 'string'}, {'type': 'integer'}]
                },
            },
        },
        'required': ['server', 'device_key'],
        'additionalProperties': False,
    }
```
items in "options" (all of them are optional) will be passed through to the request

Reply:
Oops, I didn't catch those camelcase options on the PR.

Reply:
Oh, I see. They are all direct pass-through options to the underlying api. https://github.com/Finb/bark-server/blob/master/docs/API_V2.md#push We normally do a bit more curation, turning things into snake_case, defining options as boolean, rather than the string '1' for isArchive and automaticallyCopy. Defining an enum for 'level' so that only valid options are allowed. Sorry I didn't notice this on your original PR and have the discussion there.